src/main/java/io/debezium/connector/cassandra/SchemaProcessor.java

```diff
@@ -5,24 +5,216 @@
+                LOGGER.info(String.format("Table %s.%s detected to be removed!", table.getKeyspace().getName(), table.getName()));
+                if (table.getOptions().isCDC()) {
+                    schemaHolder.removeTableSchema(new KeyspaceTable(table));
+                }
+
+                final String ksName = table.getKeyspace().getName();
+                final String tableName = table.getName();
+
+                final org.apache.cassandra.schema.KeyspaceMetadata oldKsm = Schema.instance.getKSMetaData(table.getKeyspace().getName());
+
+                if (oldKsm == null) {
+                    LOGGER.warn("KeyspaceMetadata for keyspace {} is not found!", table.getKeyspace().getName());
+                    return;
+                }
+
+                final ColumnFamilyStore cfs = Keyspace.openWithoutSSTables(ksName).getColumnFamilyStore(tableName);
+
+                if (cfs == null) {
+                    LOGGER.warn("ColumnFamilyStore for {}.{} is not found!", table.getKeyspace(), table.getName());
+                    return;
+                }
+
+                // make sure all the indexes are dropped, or else.
+                cfs.indexManager.markAllIndexesRemoved();
+
+                // reinitialize the keyspace.
+                final CFMetaData cfm = oldKsm.tables.get(tableName).get();
+                final org.apache.cassandra.schema.KeyspaceMetadata newKsm = oldKsm.withSwapped(oldKsm.tables.without(tableName));
+
+                Schema.instance.unload(cfm);
+                Schema.instance.setKeyspaceMetadata(newKsm);
+
+                LOGGER.info("Removed schema for table {}", table.asCQLQuery());
+            }
+
+            @Override
+            public void onTableChanged(final TableMetadata newTableMetadata, final TableMetadata oldTableMetaData) {
+                LOGGER.debug("Detected alternation in schema of {}.{} (previous cdc = {}, current cdc = {})",
+                        newTableMetadata.getKeyspace().getName(),
+                        newTableMetadata.getName(),
+                        oldTableMetaData.getOptions().isCDC(),
+                        newTableMetadata.getOptions().isCDC());
+
+                // if it is CDC but it was not, add it to schema
+                if (newTableMetadata.getOptions().isCDC() && !oldTableMetaData.getOptions().isCDC()) {
+                    schemaHolder.addOrUpdateTableSchema(new KeyspaceTable(newTableMetadata),
+                            new SchemaHolder.KeyValueSchema(kafkaTopicPrefix, newTableMetadata, sourceInfoStructMaker));
+                }
+                else if (!newTableMetadata.getOptions().isCDC() && oldTableMetaData.getOptions().isCDC()) {
+                    // if new table is not on cdc anymore, remove it
+                    schemaHolder.removeTableSchema(new KeyspaceTable(newTableMetadata));
+                }
+                else if (newTableMetadata.getOptions().isCDC() && oldTableMetaData.getOptions().isCDC()) {
```

Comment:
This can be combined with `if (newTableMetadata.getOptions().isCDC() && !oldTableMetaData.getOptions().isCDC())`.

Reply:
i prefer to leave it this way, it reads better in my mind instead of having these ifs embedded

Reply:
> i prefer to leave it this way, it reads better in my mind instead of having these ifs embedded

I prefer these two conditions to be combined. It might seem clearer for the person who wrote the codes, but it's confusing and redundant for other people.